File: nbui-core/src/main/java/cn/rubintry/nbui/core/config/NBElasticPullConfig.java
package cn.rubintry.nbui.core.config;

import cn.rubintry.nbui.core.INBUIConfig;
import cn.rubintry.nbui.core.NBUI;

public class NBElasticPullConfig implements INBUIConfig {

    public static final Double DEFAULT_ELASTIC_COEFFCIENT  = 0.95;

    private Double elasticCoefficient = DEFAULT_ELASTIC_COEFFCIENT;

    /**
     * 设置下拉放大阻力系数，取值范围[0,1]，默认值[0.95]，数值越大，越有弹性
     */
    public NBElasticPullConfig setElasticCoefficient(double elasticCoefficient)  {
        this.elasticCoefficient = elasticCoefficient;
        return this;
    }

    public Double getElasticCoefficient()  {
        return elasticCoefficient;
    }

    @Override
    public NBUI endConfig() {
        return NBUI.getInstance();
    }
}

File: nbui-pull/src/main/java/cn/rubintry/nbui/pull/config/NBElasticPullConfig.kt
package cn.rubintry.nbui.pull.config

import cn.rubintry.nbui.core.INBUIConfig
import cn.rubintry.nbui.core.NBUI
import cn.rubintry.nbui.pull.NBElasticView


/**
 * [cn.rubintry.nbui.pull.NBElasticView]的全局配置，优先级低于[cn.rubintry.nbui.pull.NBElasticView]中配置
 */
class NBElasticPullConfig : INBUIConfig {

    private var elasticCoefficient: Double = NBElasticView.DEFAULT_ELASTIC_COEFFCIENT

    override fun init(): NBUI {
        return NBUI.getInstance()
    }


    /**
     * 设置下拉放大阻力系数，取值范围[0,1]，默认值[0.95]，数值越大，越有弹性
     */
    fun setElasticCoefficient(elasticCoefficient: Double): NBElasticPullConfig {
        this.elasticCoefficient = elasticCoefficient
        return this
    }

    fun getElasticCoefficient(): Double {
        return elasticCoefficient
    }
}